prowler/providers/azure/services/monitor/monitor_service.py

```diff
@@ -0,0 +1,157 @@
+from dataclasses import dataclass
+
+from azure.mgmt.monitor import MonitorManagementClient
+
+from prowler.lib.logger import logger
+from prowler.providers.azure.lib.service.service import AzureService
+
+
+########################## Monitor
+class Monitor(AzureService):
+    def __init__(self, audit_info):
+        super().__init__(MonitorManagementClient, audit_info)
+        # self.activity_logs = self.__get_activity_logs__()
+        # self.activity_log_alerts = self.__get_activity_log_alerts__()
+        self.diagnostics_settings = self.__get_diagnostics_settings__()
+
+    def __get_diagnostics_settings__(self):
+        logger.info("Monitor - Getting diagnostics settings...")
+        diagnostics_settings = {}
+        for subscription, client in self.clients.items():
+            try:
+                diagnostics_settings.update({subscription: []})
+                settings = client.diagnostic_settings.list(
+                    resource_uri=f"subscriptions/{self.subscriptions[subscription]}/"
+                )
+                for setting in settings:
+                    diagnostics_settings[subscription].append(
+                        DiagnosticSetting(
+                            id=setting.id,
+                            name=setting.name,
+                            type=setting.type,
+                            event_hub_authorization_rule_id=setting.event_hub_authorization_rule_id,
+                            event_hub_name=setting.event_hub_name,
+                            metrics=setting.metrics,
+                            logs=setting.logs,
+                            workspace_id=setting.workspace_id,
+                            storage_account_id=setting.storage_account_id,
+                            service_bus_rule_id=setting.service_bus_rule_id,
+                            marketplace_partner_id=setting.marketplace_partner_id,
+                            log_analytics_destination_type=setting.log_analytics_destination_type,
```

Comment:
- Are you sure that all of these values are always present?
- Do you need all of these values for the check?

Reply:
Always try to only include the attributes you need for the checks you are coding, keep it simple.